### DISCUSSION TITLE
Maximum rate that advertisements can be scanned/read

### DISCUSSION THREAD

**Discussion by damianjwilliams** [hbldh/bleak#831] [Q&A] [answered]
Is there a maximum rate at which advertisements can be scanned? I have a peripheral broadcasting an advertisement every 20 ms, and would like to be able to read as many of the advertising packets as possible (it doesn't matter if misses a few here and there). The idea is that I am able to plot sensor data which is embedded in the advertising packet. 

My starting point is the code below, but it seems to be duplicating and missing a lot of packets so there might be a better way to do it.

```
import asyncio
from bleak import BleakScanner
from datetime import datetime

def detection_callback(device, advertisement_data):

    if device.address == "XX:XX:XX:XX:XX:XX":
        print(datetime.utcnow().strftime('%H:%M:%S.%f')[:-3])
        print(advertisement_data)

async def main():
    scanner = BleakScanner()
    scanner.register_detection_callback(detection_callback)
    await scanner.start()

while(1):
    loop = asyncio.get_event_loop()
    loop.run_until_complete(main())

```

**Comment by dlech** [comment, marked as answer]
Bleak usually only calls the callback when the advertising data changes. If the data changes in every packet, then theoretically Bleak should receive it. But the OS may handle things differently internally. There is some discussion about this in #394.

**Reply by damianjwilliams** [reply]
Thanks. I see.  As I am using Windows 10, it looks like I should find a different method because there is [no easy way of deduplication](https://github.com/hbldh/bleak/issues/394#issue-765553424) and, as far as I understand, since the [Windows 10 scan interval is fixed at 180 ms](https://stackoverflow.com/questions/37307301/ble-scan-interval-windows-10/37328965), I will [miss](https://github.com/hbldh/bleak/issues/494#issuecomment-823093591) a large number of advertising packets and effectively only be able to read advertising data at ~ 5 Hz.
